### ISSUE TITLE
What is the minimum number of training examples required for RASA to correctly perform NLU ?

### ISSUE THREAD

**Issue by madhusudhannsn** [RasaHQ/rasa_nlu#566]
I am building a personal assistant with my custom data. I have 10 intents at this moment and each intent has around 40 to 50 utterances. 

I have tried ner_crf and ner_spacy for entity extraction and I see both has their own flaws.
I have tried sklearn as intent classifier but I see it requires more training examples

> UndefinedMetricWarning: F-score is ill-defined and being set to 0.0 in labels with no predicted samples.

I see MITIE is taking more than 10-12 hours for training a model, even I use more threads

> "num_threads": 4,
>   "max_training_processes": 4,

As I am new to RASA, can some one please recommend me what backend I can use for production application and what amount of training examples required for building such application ?

**Comment by tmbo** [comment]
@madhusudhannsn The amount of training data highly depends on your domain (e.g. how close in meaning and structure the different intents are). 40-50 sounds reasonable though. The spacy warning is indicating that a bit more training data might be good, but you might be good already. Try splitting your training data into two files (one for training and one for testing). You can run it with 

```
python -m rasa_nlu.evaluate -d my_evaluation_data.json -c spacy_sklearn_config.json -m models/model_20170901_122312
```
(parameters are just examples). It will evaluate the model with respect to the intent classifcation.

**Comment by tmbo** [comment]
In general, you should be fine using `spacy_sklearn` (which is preferable to `mitie` due to the training time - model performance is usually nearly the same)

**Comment by madhusudhannsn** [comment]
@tmbo Thanks for your suggestions. I was missing the evaluate step. I hope that improves the intent classification process.
